src/views/ProductDetail.vue

```diff
@@ -97,43 +69,151 @@
+import { StockService } from '@/services/StockService'
+import { hasError, getFeature, showToast } from "@/utils";
+import { sortSizes } from '@/apparel-sorter';
+import OtherStoresInventoryModal from "./OtherStoresInventoryModal.vue";
+import { translate } from "@/i18n";
 
 export default defineComponent({
   name: "ProductDetail",
   components: {
     IonBackButton,
     IonButton,
-    IonCard,
     IonChip,
     IonContent,
     IonHeader,
     IonItem,
-    IonItemDivider,
     IonLabel,
     IonList,
     IonListHeader,
     IonNote,
     IonPage,
+    IonRow,
     IonTitle,
     IonToolbar,
     Image
   },
+  data() {
+    return {
+      selectedColor: '',
+      selectedSize: '',
+      features: [] as any,
+      currentVariant: {} as any,
+      currentStoreInventory: 0,
+      otherStoresInventory: 0,
+      warehouseInventory: 0,
+      otherStoresInventoryDetails: [] as any
+    }
+  },
+  computed: {
+    ...mapGetters({
+      product: "product/getCurrent",
+      currentFacility: 'user/getCurrentFacility',
+    })
+  },
+  async beforeMount() {
+    await this.store.dispatch('product/getVariants', { productId: this.$route.params.productId })
+    this.getFeatures()
+    await this.updateVariant()
+  },
+  methods: {
+    async applyFeature(feature: string, type: string) {
+      if(type === 'color') this.selectedColor = feature;
+      else if(type === 'size') this.selectedSize = feature
+      await this.updateVariant();
+    },
+    getFeatures() {
+      const features = {} as any
+      this.product.variants.map((variant: any) => {
+        const size = getFeature(variant.featureHierarchy, '1/SIZE/');
+        const color = getFeature(variant.featureHierarchy, '1/COLOR/');
+        if (!features[color]) features[color] = [size];
+        else if(!features[color].includes(size)) features[color].push(size);
+      });
+
+      Object.keys(features).forEach((color) => this.features[color] = sortSizes(features[color]))
+
+      this.selectedColor = Object.keys(this.features)[0];
+      this.selectedSize = this.features[this.selectedColor][0];
+    },
+    async updateVariant() {
+      let variant;
+      if(this.selectedColor || this.selectedSize) {
+        variant = this.product.variants.find((variant: any) => {
+          const hasSize = getFeature(variant.featureHierarchy, '1/SIZE/') === this.selectedSize;
+          const hasColor = getFeature(variant.featureHierarchy, '1/COLOR/') === this.selectedColor;
+          if (hasSize && hasColor) return variant;
+        })
+
+        if(!variant) {
+          this.selectedSize = this.features[this.selectedColor][0];
+          variant = this.product.variants.find((variant: any) => getFeature(variant.featureHierarchy, '1/SIZE/') === this.selectedSize)
+          showToast(translate("Selected variant not available"));
+        }
+      }
+      this.currentVariant = variant;
+      this.currentStoreInventory = this.otherStoresInventory = this.warehouseInventory = 0;
```

Comment:
This could be handled in check inventory

Reply:
Updated